### PR TITLE
Add units to JSON request

### PR DESCRIPTION
**Pull request by dnesbitt61**
Change to floating point (not int meters)

### CODE REVIEW

**Comment by dgearhart** [comment]
:shipit:

File: src/tyr/custom_route_handler.cc [hunk 1]
```diff
@@ -93,13 +94,14 @@ json::MapPtr summary(const valhalla::odin::TripDirections& trip_directions){
...
+  float length = 0.0f;
   uint64_t seconds = 0, meters = 0;
   for(const auto& maneuver : trip_directions.maneuver()) {
-    meters += static_cast<uint64_t>(maneuver.length() * 1000.f);
+    length += maneuver.length();
     seconds += static_cast<uint64_t>(maneuver.time());
   }
   route_summary->emplace("time", seconds);
-  route_summary->emplace("distance", meters);
+  route_summary->emplace("length", (long double)(length));
```
Comment (kevinkreiser) [inline src/tyr/custom_route_handler.cc]: c style cast

File: src/tyr/custom_route_handler.cc [hunk 2]
```diff
@@ -158,7 +161,7 @@ json::ArrayPtr legs(const valhalla::odin::TripPath& trip_path,
     if (street_names->size())
       man->emplace("streetNames", street_names);
     man->emplace("time", static_cast<uint64_t>(maneuver.time()));
-    man->emplace("distance", static_cast<uint64_t>(maneuver.length() * 1000.f));
+    man->emplace("length", (long double)(maneuver.length()));
```
Comment (kevinkreiser) [inline src/tyr/custom_route_handler.cc]: again

File: src/tyr/custom_route_handler.cc [hunk 3]
```diff
@@ -178,7 +181,7 @@ json::ArrayPtr legs(const valhalla::odin::TripPath& trip_path,
   }
   leg->emplace("maneuvers", maneuvers);
   summary->emplace("time", seconds);
-  summary->emplace("distance", meters);
+  summary->emplace("length", (long double)length);
```
Comment (kevinkreiser) [inline src/tyr/custom_route_handler.cc]: again :o)

File: src/tyr/custom_route_handler.cc [hunk 4]
```diff
@@ -238,6 +244,21 @@ CustomRouteHandler::CustomRouteHandler(const boost::property_tree::ptree& config
   //get the config for the graph reader
   reader_.reset(new valhalla::baldr::GraphReader(config.get_child("mjolnir.hierarchy")));
 
+  // Get the units (defaults to kilometers
+  std::string units = "kilometers";
+  try {
+    units = request.get<std::string>("units");
+  }
+  catch(...) {
+    std::cout << "No units string - assume kilometers" << std::endl;
```
Comment (kevinkreiser) [inline src/tyr/custom_route_handler.cc]: we dont log with cout

File: valhalla/tyr/handler.h [hunk 5]
```diff
@@ -36,7 +36,8 @@ class Handler {
   virtual std::string Action() = 0;
 
  protected:
-
+  bool km_units_;
```
Comment (kevinkreiser) [inline valhalla/tyr/handler.h]: definitely doesnt belong in the base class, should go in the derived one

**Pull request merged**